Python/getargs.c

```diff
@@ -2586,7 +2586,7 @@ _PyArg_UnpackKeywordsWithVararg(PyObject *const *args, Py_ssize_t nargs,
             current_arg = NULL;
         }
 
-        buf[i + vararg + 1] = current_arg;
+        buf[i + 1] = current_arg;
```

Comment:
Is there a case where the problem in this change can be triggered?

Reply:
>To see if the OOB bug is fixed in commit https://github.com/python/cpython/commit/620396279cf94e09bd7cd6e925e6e80471130d6f , I should write some code snippet calling function _PyArg_UnpackKeywordsWithVararg, compile it and ensure that it does not crash.
>
> I'm not sure how can I do this without touching the Python vm or library,
though I have wrote a proof-of-concept by adding a built-in function and test it manually:
https://github.com/colorfulappl/cpython/tree/unpack_keywords_oob_bug_poc .


I have written a PoC as a built-in function here:
https://github.com/colorfulappl/cpython/commit/8e7aa16b093fe027c9fd9a8e0f3f9b329cd507b3

You can build it and type
```python
oob_poc(1, 2, 3, 4, kw1=5, kw2=6)
```
to trigger this bug.

Reply:
@kumaraditya303, have you started working on #96002 yet? This test would be a good candidate.

Reply:
I haven't started yet, feel free to take over it.

Reply:
I don't have time for it right now; my schedule is full this weekend. Possibly next week, but I can't promise anything.

Reply:
@erlend-aasland I'm willing to write a draft this weekend, if you don't mind.

Reply:
Sure, go ahead!

Reply:
> I'm willing to write a draft this weekend, if you don't mind.

FTR, the PR is up: gh-96178

Reply:
It looks to me like we can mark this conversation as resolved.